nav2_navfn_planner/src/navfn_planner.cpp

```diff
@@ -213,26 +213,35 @@ NavfnPlanner::makePlan(
 
   double resolution = costmap_->getResolution();
   geometry_msgs::msg::Pose p, best_pose;
-  p = goal;
 
   bool found_legal = false;
-  double best_sdist = std::numeric_limits<double>::max();
-
-  p.position.y = goal.position.y - tolerance;
-
-  while (p.position.y <= goal.position.y + tolerance) {
-    p.position.x = goal.position.x - tolerance;
-    while (p.position.x <= goal.position.x + tolerance) {
-      double potential = getPointPotential(p.position);
-      double sdist = squared_distance(p, goal);
-      if (potential < POT_HIGH && sdist < best_sdist) {
-        best_sdist = sdist;
-        best_pose = p;
-        found_legal = true;
+
+  p = goal;
+  double potential = getPointPotential(p.position);
+  if (potential < POT_HIGH) {
```

Comment:
Shouldn't this be  taken care of from the while loop condition (p.position.y <= goal.position.y + tolerance)? In that case that its the goal, it meets that spec and it shouldn't even enter the while loop